view_conversations/conv-viewer-webapi/src/routes.ts

```diff
@@ -69,16 +70,52 @@ export function setup(
     }
   });
 
+  app.get('/api/toxicity/:score', async (req, res) => {
+    try {
+      let score: number = req.params.score;
+      const index = conf.spannerTableName + '_by_toxicity';
+
+      // TODO remove outer try wrapper unless it get used.
+      const sqlQuery = `SELECT *
+             FROM ${table}@{FORCE_INDEX=${index}}
+             WHERE RockV6_1_TOXICITY <= ${score} and type != "DELETION"
```

Comment:
You need to sanitize/force the input parameters to be SQL-safe. You don't want end users being able to do SQL injections attacks.

Reply:
Added a type validation for req.params.score, limited it to float type.

Reply:
cool. I think it's better to move the asserting into a simple stand-alone function e.g. "assertStrIsFloat", and then add a test for that. It's important to get that stuff right.